.github/workflows/build-pr.yml

```diff
@@ -0,0 +1,54 @@
+name: Build PR
+
+on:
+  pull_request:
+    branches:
+    - main
+
+env:
+  MAJOR: 4
+  MINOR: 0
+  MINORMINOR: 1 
+  RUN: ${{ github.run_number }}
+
+jobs:
+  build:
+    runs-on: windows-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v2
+
+      - name: Restore dependencies
+        run: |
+          nuget sources add -username Open-Systems-Pharmacology -password ${{ secrets.GITHUB_TOKEN }} -name OSP-GitHub-Packages -source "https://nuget.pkg.github.com/Open-Systems-Pharmacology/index.json"
+          nuget restore
+
+      - name: define env variables
+        run: |
+          echo "APP_VERSION=${{ env.MAJOR }}.${{ env.MINOR }}.${{ env.MINORMINOR }}.${{ env.RUN }}" | Out-File -FilePath $env:GITHUB_ENV -Append
+
+      - name: Build
+        run: msbuild OSPSuite.BDDHelper.sln /p:Version=${{env.APP_VERSION}}
+
+      - name : Test
+        run: dotnet test .\tests\**\bin\Debug\net8\OSPSuite*Tests.dll -v normal --no-build  --logger:"html;LogFileName=../testLog_Windows.html"
+
+      - name: Pack the project
+        run: dotnet pack .\OSPSuite.BDDHelper.sln --no-build --no-restore -o ./ -p:PackageVersion=${{env.APP_VERSION}} --configuration=Debug --no-build
+
+      - name: Push nupkg as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: OSPSuite.BDDHelper
+          path: ./*.nupkg
```

Comment:
This part belongs to another workflow (on push)

Reply:
forget it

Reply:
The push workflow publishes the artifact as a package that can be consumed by nuget. This one just builds it and uploads as an artifact of the build, but you can't target it with nuget.

This is just so it can be inspected